src/backend/distributed/planner/multi_explain.c

```diff
@@ -639,6 +682,192 @@ BuildRemoteExplainQuery(char *queryString, ExplainState *es)
 }
 
 
+bool
+ShouldSaveQueryExplain(QueryDesc *queryDesc, int eflags)
+{
+	PlannedStmt *plannedStmt = queryDesc->plannedstmt;
+
+	return SaveTaskExplainPlans &&
+		   ExecutorLevel == 0 &&
+		   !IsParallelWorker() &&
+		   (eflags & EXEC_FLAG_EXPLAIN_ONLY) == 0 &&
+		   !IsCitusPlan(plannedStmt->planTree);
+}
+
+
+void
+SaveQueryExplain(QueryDesc *queryDesc)
+{
+	/*
+	 * Make sure stats accumulation is done.  (Note: it's okay if several
+	 * levels of hook all do this.)
+	 */
+	InstrEndLoop(queryDesc->totaltime);
+
+	ExplainState *es = NewExplainState();
+
+	es->analyze = true;
+	es->verbose = TaskExplainOptions.verbose;
+	es->buffers = TaskExplainOptions.buffers;
+	es->timing = TaskExplainOptions.timing;
+	es->summary = TaskExplainOptions.summary;
+	es->format = TaskExplainOptions.format;
+	es->costs = TaskExplainOptions.costs;
+
+	ExplainBeginOutput(es);
+
+	if (TaskExplainOptions.query)
+	{
+		ExplainQueryText(es, queryDesc);
+	}
+
+	ExplainPrintPlan(es, queryDesc);
+
+	if (es->costs)
+	{
+		ExplainPrintJITSummary(es, queryDesc);
+	}
+
+	ExplainEndOutput(es);
+
+	MemoryContext oldContext = MemoryContextSwitchTo(TopMemoryContext);
+
+	SavedExplainPlan = makeStringInfo();
```

Comment:
Why not just `pstrdup(es->str->data)` with `SavedExplainPlan` as `char *`?

Reply:
`RemoteExplainPlan->explainOutputList` (where this eventually will end up) is list of StringInfo, so we need to convert it to StringInfo at some point if not here.